packages/icons-vue/tasks/createIconComponent.js

```diff
@@ -58,6 +59,23 @@ export default {
       svgData.class = svgData.class || {}; // may be no static class
       svgData.class[data.class] = true;
     }
+
+    // combine incoming staticStyle, style and internal style attribute.
+    const style = svgData.attrs.style;
+    delete svgData.attrs.style;
+
+    // convert incoming style attr to object form.
+    const styleArray = style.split(';').filter(item => item.trim().length > 0);
+    const styleObj = styleArray.reduce((acc, styleString) => {
```

Comment:
Would you want to do this at build time versus compile time? In our base component for React we end up just building the object in scope:

https://github.com/carbon-design-system/carbon/blob/master/packages/icon-build-helpers/src/builders/react/components/Icon.js#L12-L14

This value used to derive from the value coming from icon-helpers (e.g. we parsed the string value) but now we just inline it to make things easier.

Reply:
Is this what you were suggesting @joshblack ?